docs/design/enhance-pluggable-policy.md

```diff
@@ -0,0 +1,175 @@
+# Enhance Pluggable Policy
+While trying to develop an authorization plugin for Kubernetes, we found a few places where API extensions would ease development and add power.  There are a few goals:
+ 1.  Provide an authorization plugin that can evaluate a .Authorize() call based on the full content of the request to RESTStorage.  This includes information like the full verb, the content of creates and updates, and the names of resources being acted upon.
+ 1.  Provide a mechanism for a user to determine whether or not he can take an action without attempting the action itself.
+ 1.  Provide a way for disconnected pieces of the system to ask whether a user is permitted to take an action without running in process with the API Authorizer.  For instance, a proxy for exec calls could ask whether a user can run the exec they are requesting.
+ 1.  Provide a mechanism to discover who can perform a given action on a given resource.  This is useful for answering questions like, "who can create replication controllers in my namespace".
```

Comment:
> clarify that this is a REST API?

I've collapsed this down, deciding that one covers the other.  I haven't indicated that this is REST API, because if the `Authorizer` conforms to the interface, the REST API simply exposes this access without the provider having to do anything.